### PR TITLE
Add experimental support for Java 17 class files

### PR DESCRIPTION
**Pull request by marchof**

### CODE REVIEW

**Comment by marchof** [comment]
Sounds a bit early but this way we can continue testing against the latest JDK version in [openjdk/jdk](https://github.com/openjdk/jdk) repo.

File: org.jacoco.core.test.validation/pom.xml [hunk 1]
```diff
@@ -257,6 +257,31 @@
...
+        <!-- see respective profile in org.jacoco.build about this override -->
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+      </properties>
+      <modules>
+        <module>../org.jacoco.core.test.validation.kotlin</module>
+        <module>../org.jacoco.core.test.validation.java7</module>
+        <module>../org.jacoco.core.test.validation.java8</module>
+        <module>../org.jacoco.core.test.validation.java14</module>
+        <!-- Does not yet support Java 17
```
Comment (Godin) [inline org.jacoco.core.test.validation/pom.xml:279]:
```suggestion
        <!-- Groovy 3.0.7 does not support Java 17
```

**Review by Godin — APPROVED** [review]

**Comment by garydgregory** [comment]
Hi, When can we expect 0.8.7? FWIW, over at Apache Commons, we are using Java 17-EA in our GitHub builds to try to catch problems early. TY.

**Comment by marchof** [comment]
Please see out [FAQ](https://www.jacoco.org/jacoco/trunk/doc/faq.html)

> When will feature X be released?
> 
> JaCoCo is maintained by volunteers in their free time. Since we cannot guarantee free capacity, we do not commit to
> particular release dates. Typically, you can expect a couple of releases every year.
>
> In the change log, you can see all features that have been implemented in master branch and will be available with the next
> release. And in the meantime you can test latest build of master branch (Maven SNAPSHOT) and provide feedback to us.

We actually do [test against all JDKs from 5 to 17](https://dev.azure.com/jacoco-org/JaCoCo/_build/latest?definitionId=1&branchName=master).

**Comment by garydgregory** [comment]
> Please see out [FAQ](https://www.jacoco.org/jacoco/trunk/doc/faq.html)
> 
> > When will feature X be released?
> > JaCoCo is maintained by volunteers in their free time. Since we cannot guarantee free capacity, we do not commit to
> > particular release dates. Typically, you can expect a couple of releases every year.
> > In the change log, you can see all features that have been implemented in master branch and will be available with the next
> > release. And in the meantime you can test latest build of master branch (Maven SNAPSHOT) and provide feedback to us.
> 
> We actually do [test against all JDKs from 5 to 17](https://dev.azure.com/jacoco-org/JaCoCo/_build/latest?definitionId=1&branchName=master).

Yep, I have a stock "we are all volunteers" response I also give folks over at Apache Commons and Apache Log4j ;-)

Still, it would be nice to vaguely know when we can expect to get a release to run on Java 16/17 EAs.

**Comment by marchof** [comment]
@garydgregory Sorry, no offense :)

In any case there will be a official JaCoCo release with Java 17 support before Java 17 is released -- which is scheduled for 2021/09. But I assume we will have a release earlier as soon as ASM has official support for Java 16.

**Pull request merged**